sdcm/utils/health_checker.py

```diff
@@ -173,3 +173,30 @@ def check_schema_version(gossip_info, peers_details, nodes_status, current_node)
                                     node=current_node.name,
                                     message=f'Current node {current_node.ip_address}. Schema version is not same on all '
                                     f'nodes in SYSTEM.PEERS info: {peers_info_str}')
+
+
+def check_schema_agreement(node_name, node_ip_address, gossip_info, peers_info):
+    message_pref = f'Check for schema agreement on the node {node_name} ({node_ip_address}).'
+    LOGGER.debug(f'{message_pref} Gossip info: {gossip_info}')
+
+    gossip_schema = {data["schema"] for data in gossip_info.values() if data['status'] == "NORMAL"}
+    if len(gossip_schema) > 1:
+        LOGGER.warning(f'{message_pref} Schema version is not same on nodes in the gossip')
+        return False
+
+    LOGGER.debug(f'{message_pref} Peers info: {peers_info}')
+
+    peers_schema = set()
+    for ip, data in gossip_info.items():
+        if data['status'] == "NORMAL" and ip in peers_info:
+            if data["schema"] != peers_info[ip]['schema_version']:
+                LOGGER.warning(f'{message_pref} Schema version is not same in the gossip and peers for {ip}')
+                return False
+            peers_schema.add(peers_info[ip]['schema_version'])
+
+    if len(peers_schema) > 1:
+        LOGGER.warning(f'{message_pref} Schema version is not same for all peers: {peers_schema}')
+        return False
```

Comment:
I performed it on such way for point exactly on the node with different schema version, not just say "it's not same on all nodes". So when we investigate the test results, we won't need to compare manually (by the eye, it's not so trivial) of schema versions between gossip and peers and search which node have the problem. The message will keep all information

Reply:
Think again: on previous step you discovered that all `data["schema"]` with `data['status'] == "NORMAL"` are equal. So, there is no way for code in `if len(peers_schema) > 1:` block to be executed.

Reply:
I see. I changed it. I still need go over gossip list because of I want to filter out the nodes that not in NORMAL status and there is no such info in the peers table.